WcaOnRails/spec/models/user_spec.rb

```diff
@@ -353,6 +353,27 @@
     end
   end
 
+  describe "check if registration form sends mail to newly registered user" do
+    it "send mail" do
+      new_user = FactoryBot.create :user, :unconfirmed
+      expect(ActionMailer::Base.deliveries.last.to).to eq [new_user.email]
+      expect(ActionMailer::Base.deliveries.last.subject).to eq "Confirmation instructions"
+    end
+  end
+
+  describe "let user change his mail address" do
+    it "send mail" do
+      user = FactoryBot.create :user
+      user.unconfirmed_email = "new@example.com"
+      expect(user.pending_reconfirmation?).to eq true
+      user.send_reconfirmation_instructions
+
+      expect { user.send_reconfirmation_instructions } .to change { ActionMailer::Base.deliveries.count }.by(1)
+      expect(ActionMailer::Base.deliveries.last.to).to eq [user.unconfirmed_email]
```

Comment:
As I commented above, I think it would be better to check a mock here, rather than looking at the actual email that was sent.